### PR TITLE
PUBDEV-8871: XGBoost - disable full scoring and use eval_metric instead

### PR DESCRIPTION
**Pull request by michalkurka**

### CODE REVIEW

File: h2o-core/src/main/java/hex/ScoreKeeper.java [hunk 1]
```diff
@@ -466,6 +469,7 @@ public String toString() {
         ", _hitratio=" + Arrays.toString(_hitratio) +
         ", _lift=" + _lift +
         ", _anomaly_score_normalized=" + _anomaly_score_normalized +
+        ", _custom=" + _custom_metric +
```
Comment (kanech) [inline h2o-core/src/main/java/hex/ScoreKeeper.java:472]: Should the text correspond to the field name? I.e. 
` ", _custom_metric=" + _custom_metric +`

File: h2o-py/tests/testdir_algos/xgboost/pyunit_xgboost_eval_metric_stopping.py [hunk 2]
```diff
@@ -35,8 +36,25 @@ def test_eval_metric_early_stopping():
...
+
 
-if __name__ == "__main__":
-    pyunit_utils.standalone_test(test_eval_metric_early_stopping)
-else:
-    test_eval_metric_early_stopping()
+pyunit_utils.run_tests([
+    test_eval_metric_early_stopping,
+    test_eval_metric_early_stopping_native_scoring_only
+])
```
Comment (kanech) [inline h2o-py/tests/testdir_algos/xgboost/pyunit_xgboost_eval_metric_stopping.py:60]: Why this test file doesn't follow the 
`if __name__ == "__main__": 
...
else:
...
`
structure as all other test files in this folder do?

File: h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java [hunk 3]
```diff
@@ -144,6 +144,7 @@ public enum Backend {
     public Backend _backend = Backend.auto;
 
     public String _eval_metric;
+    boolean _score_eval_metric_only;
```
Comment (kanech) [inline h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java:147]: Some leftover?

File: h2o-algos/src/main/java/hex/tree/SharedTree.java [hunk 4]
```diff
@@ -974,7 +974,8 @@ public static TwoDimTable createScoringHistoryTable(Model.Output _output,
 
     int rows = 0;
     for( int i = 0; i<_scored_train.length; i++ ) {
-      if (i != 0 && Double.isNaN(_scored_train[i]._rmse) && (_scored_valid == null || Double.isNaN(_scored_valid[i]._rmse))) continue;
+      if (i != 0 && Double.isNaN(_scored_train[i]._rmse) && Double.isNaN(_scored_train[i]._custom_metric) && 
+              (_scored_valid == null || (Double.isNaN(_scored_valid[i]._rmse) && Double.isNaN(_scored_valid[i]._custom_metric)))) continue;
       rows++;
```
Comment (kanech) [inline h2o-algos/src/main/java/hex/tree/SharedTree.java:979]: Would it make sense to wrap the complicated (and duplicated) conditional logic into a helper method that would provide a clue to a casual reader what's going on? E.g.
`if (i != 0 && isScoreMissingAtRow(i)) continue;`

**Review by kanech — APPROVED** [review]
I left some comments in places where I wasn't sure if they are correct. Please revisit them and if there is no problem then it's OK to merge these changes.

File: h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java (hunk 3 above)
Comment (michalkurka) [reply h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java:147]: This parameter is needed but it should be public - thanks for the comment.

File: h2o-core/src/main/java/hex/ScoreKeeper.java (hunk 1 above)
Comment (michalkurka) [reply h2o-core/src/main/java/hex/ScoreKeeper.java:472]: yeah, sure - it is better for consistency to keep the print out the same - will fix, thank you

Comment (michalkurka) [reply h2o-core/src/main/java/hex/ScoreKeeper.java]: fixed

File: h2o-py/tests/testdir_algos/xgboost/pyunit_xgboost_eval_metric_stopping.py (hunk 2 above)
Comment (michalkurka) [reply h2o-py/tests/testdir_algos/xgboost/pyunit_xgboost_eval_metric_stopping.py:60]: `run_tests` was added later than most of these tests, it is a newer mechanism and is better when you have multiple test cases in the same file. Eg. StackEnsemble tests almost exclusively use run_tests. There is not loss of functionality.

Thanks for the comment, good to check!

File: h2o-algos/src/main/java/hex/tree/SharedTree.java (hunk 4 above)
Comment (michalkurka) [reply h2o-algos/src/main/java/hex/tree/SharedTree.java:979]: yes, you are right - let me see if I can improve it

Comment (michalkurka) [reply h2o-algos/src/main/java/hex/tree/SharedTree.java:978]: I will just use isEmpty from ScoreKeeper - in this context it should have the same meaning effectively

**Comment by michalkurka** [comment]
Thank you @kanech for the comments. I incorporated your suggestions.

**Pull request merged**